operators/pkg/controller/elasticsearch/version/zen1/minimum_masters.go

```diff
@@ -96,19 +94,40 @@ func UpdateMinimumMasterNodes(
 		return true, nil
 	}
 
+	return false, UpdateMinimumMasterNodesTo(es, esClient, actualStatefulSets, reconcileState, minimumMasterNodes)
+}
+
+// UpdateMinimumMasterNodesTo calls the ES API to update the value of zen1 minimum_master_nodes
+// to the given value, if the cluster is using zen1.
+func UpdateMinimumMasterNodesTo(
+	es v1alpha1.Elasticsearch,
+	esClient client.Client,
+	actualStatefulSets sset.StatefulSetList,
+	reconcileState *reconcile.State,
+	minimumMasterNodes int,
+) error {
+	if !AtLeastOneNodeCompatibleWithZen1(actualStatefulSets) {
```

Comment:
We already have that check in both callers of this func. Should we either remove it from there or remove it from here?

Reply:
You're right. Let's keep it here and remove it from other places?
I like to have the check done in the `zen1` package. The flow being to always call zen1 stuff in higher-level functions, which may end up be a no-op, instead of having higher-level functions checking if they need to call zen1 or not.
WDYT?

Reply:
Sounds good.

Reply:
Hm actually `UpdateMinimumMasterNodes` looks like this:

```go

// This is the call we'd like to remove
if !AtLeastOneNodeCompatibleWithZen1(actualStatefulSets) {
		// nothing to do
		return false, nil
	}

// ...

if currentAvailableMasterCount < minimumMasterNodes {
		// This is expected to happen from time to time
		log.V(1).Info("Not enough masters to update the API",
			"namespace", es.Namespace,
			"es_name", es.Name,
			"current", currentAvailableMasterCount,
			"minimum_master_nodes", minimumMasterNodes)

// this is also checking AtLeastOneNodeCompatibleWithZen1()
return UpdateMinimumMasterNodesTo(es, esClient, actualStatefulSets, reconcileState, minimumMasterNodes)
```

If I remove the first few lines since the call is already done later, we get into a situation where we log something that does not make much sense if zen1 is not applicable.

So I think I'll keep it here.
But still remove the duplicated call in `downscale.go/maybeUpdateZen1ForDownscale` as you suggested.

Reply:
Actually, emitting an event on zen1 2->1 scenarios also kind of forces me to call `AtLeastOneNodeCompatibleWithZen1 ` in `downscale.go/maybeUpdateZen1ForDownscale `, so the event is not emitted if the cluster is using zen2...